src/components/CommentCard.jsx

```diff
@@ -1,26 +1,50 @@
 import { useState, useEffect } from "react";
 import { formatDate } from "../utils/formatDate";
+import { ClipLoader } from "react-spinners";
 import { BiLike } from "react-icons/bi";
 import * as api from "../utils/api";
 
-const CommentCard = ({comment}) => {
-    const [user, setUser] = useState();
-    const { author, created_at, votes, body } = comment;
+const CommentCard = ({ comment, currUser, setComments, setCommentDeleted }) => {
+    const [commentAuthor, setCommentAuthor] = useState();
+    const [deleteError, setDeleteError] = useState(false);
+    const [isDeleting, setIsDeleting] = useState(false);
+    const { author, created_at, votes, body, comment_id } = comment;
 
     useEffect(() => {
         api.getUserByUsername(author).then((data) => {
-            setUser(data.user);
+            setCommentAuthor(data.user);
         })
     }, [author])
 
+    const deleteComment = (commentId) => {
+        setIsDeleting(true);
+        api.deleteCommentByCommentId(commentId).then((data) => {
+            setCommentDeleted(true);
+            setDeleteError(false);
+            setIsDeleting(false);
+            setComments((currComments) => {
+                return currComments.filter((comment) => {
+                    return comment.comment_id !== commentId;
+                })
+            })
+        }).catch((err) => {
+            setIsDeleting(false);
+            setCommentDeleted(false);
+            setDeleteError(true);
+        })
```

Comment:
Could you either lock the delete button, or replace it with the loading thing when you click it, to stop it being clicked multiple times on a slow connection? I can spam it loads of times at the moment if the internet is slow.